PrePi/FdtParser.c

```diff
@@ -43,10 +56,24 @@ FindMemnodeFdt (
 
   //
   // Retrieve the #address-cells and #size-cells properties
-  // from the root node, or use the default if not provided.
+  // from the memory or root node, or use the default if not provided.
   //
   AddressCells = 1;
   SizeCells = 1;
+  AddressCellsRoot = -1;
+  SizeCellsRoot = -1;
+
+  Prop = fdt_getprop (DeviceTreeBlob, RootNode, "#address-cells", &Length);
+  if (Length == 4) {
+    AddressCells = fdt32_to_cpu (*Prop);
+    AddressCellsRoot = AddressCells;

```

Comment:
at this point AddressCells == AddressCellsRoot, I don't get the reason.

Reply:
The reason is that this way it does not have to be copied later if the local value is not found